src/txmempool.cpp

```diff
@@ -469,11 +469,43 @@ void CTxMemPool::removeUnchecked(txiter it)
     cachedInnerUsage -= it->DynamicMemoryUsage();
     cachedInnerUsage -= memusage::DynamicUsage(mapLinks[it].parents) + memusage::DynamicUsage(mapLinks[it].children);
     mapLinks.erase(it);
+    if (!minedInBlock) {
+        if (trackRemovalsCount) {
+            removedTxs.emplace_back(it->GetSharedTx());
+        }
+        else {
+            // If there are ever intentionally removals that are not
+            // meant to be tracked (so they can be notified on), then
+            // this log message can be removed.
+            LogPrint("mempool", "MemPoolRemovalTracker not tracking removed transaction %s\n",hash.ToString());
```

Comment:
about the error?  this is the only log of the fact that a removal happened without notification.